Extensions/timestamps.js

```diff
@@ -230,6 +265,89 @@ XKit.extensions.timestamps = new Object({
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $(XKit.extensions.timestamps.posts_class).not(".xkit_timestamps");
+
+		if (!posts || posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
+
+		posts.each(function() {
+			var post = $(this);
+
+			post.addClass("xkit_timestamps");
+
+			// What is the react class for .fan_mail?
+//			if (post.hasClass("fan_mail")) {
+//				return;
+//			}
+
+			// What is the react id for #new_post?
+//			if (post.attr('id') === "new_post" || post.find(XKit.extensions.timestamps.private_posts_class).length > 0)
+			if (post.find(XKit.extensions.timestamps.private_posts_class).length > 0) {
+				return;
+			}
+
+			var post_id = $(this).parents("[data-id]").attr("data-id");
+
+			var xtimestamp_class = "xtimestamp";
+			if (XKit.extensions.timestamps.in_search && !$("#search_posts").hasClass("posts_view_list")) {
+				xtimestamp_class = "xtimestamp-in-search";
+			}
+
+			var xtimestamp_html = `<div class="xkit_timestamp_${post_id} ${xtimestamp_class} xtimestamp_loading">&nbsp;</div>`;
+			post.find(XKit.extensions.timestamps.meatballs_class).prepend(xtimestamp_html);
```

Comment:
nice try, but we should leave the header alone and replicate the timestamps behaviour that everyone's used to. the timestamps should be inserted into the post after the `header` element.

when creating this new block element, you'll want to use the CSS variable `--post-padding` for the left/right padding, `--post-header-vertical-padding` for the bottom margin, and force the element's height to exist with `line-height: 1em`. some negative top margin would look nice too

my mockup:
![image](https://user-images.githubusercontent.com/28949509/83654420-00d76080-a5b5-11ea-9d14-97f5f291db51.png)

CSS used:
```css
font-size: 14px;
line-height: 1em;
padding: 0 var(--post-padding);
margin-top: calc(var(--post-header-vertical-padding) / -2);
margin-bottom: var(--post-header-vertical-padding);
```